### PR TITLE
Fix: Correcting map location

### LINKED ISSUE #1219: Map showing wrong location
**Actual Behaviour**

These is location (0,0) .
screenshot_20170302-020225

**Expected Behaviour**

screenshot_20170302-020248

**Steps to reproduce it**

Open map from session detail  activity.

**Would you like to work on the issue?**
yes

### PR DESCRIPTION
**Pull request by chiragw15**
Fixes #1219 
Changes: Corrected map location. Checked if microlocation is (0,0) i.e if latitude or longitude of microlocation is not given in api, then displayed event location.

### CODE REVIEW

File: android/app/src/fdroid/java/org/fossasia/openevent/fragments/OSMapFragment.java [hunk 1]
```diff
@@ -92,9 +92,16 @@ private void populateMap() {
                 Session session =  DbSingleton.getInstance().getSessionById(id);
                 int location_id = session.getMicrolocation().getId();
                 Microlocation microlocation =  DbSingleton.getInstance().getMicrolocationById(location_id);
-                setDestinationLatitude(microlocation.getLatitude());
-                setDestinationLongitude(microlocation.getLongitude());
-                setDestinationName(microlocation.getName());
+                if( microlocation.getLatitude() == 0 && microlocation.getLongitude() == 0) {
```
Comment (harshitagupta30) [inline android/app/src/fdroid/java/org/fossasia/openevent/fragments/OSMapFragment.java]: reformat the statement.

File: android/app/src/fdroid/java/org/fossasia/openevent/fragments/OSMapFragment.java [hunk 2]
```diff
@@ -92,9 +92,16 @@ private void populateMap() {
...
+                    Event event = DbSingleton.getInstance().getEventDetails();
+                    setDestinationLatitude(event.getLatitude());
+                    setDestinationLongitude(event.getLongitude());
+                    setDestinationName(event.getLocationName());
+                } else {
+                    setDestinationLatitude(microlocation.getLatitude());
+                    setDestinationLongitude(microlocation.getLongitude());
+                    setDestinationName(microlocation.getName());
+                }
             }else{
```
Comment (harshitagupta30) [inline android/app/src/fdroid/java/org/fossasia/openevent/fragments/OSMapFragment.java]: reformat the condition.

**Comment by mananwason** [comment]
Please undo the refactoring. It's very time consuming to check the changes here.

**Comment by chiragw15** [comment]
Done @mananwason

**Comment by mananwason** [comment]
LGTM

**Pull request merged**